action/protocol/execution/evm/evm.go

```diff
@@ -287,12 +288,21 @@ func ExecuteContract(
 	}
 	stateDB.clear()
 
-	if featureCtx.SetRevertMessageToReceipt && receipt.Status == uint64(iotextypes.ReceiptStatus_ErrExecutionReverted) && retval != nil && bytes.Equal(retval[:4], _revertSelector) {
-		// in case of the execution revert error, parse the retVal and add to receipt
-		data := retval[4:]
-		msgLength := byteutil.BytesToUint64BigEndian(data[56:64])
-		revertMsg := string(data[64 : 64+msgLength])
-		receipt.SetExecutionRevertMsg(revertMsg)
+	if featureCtx.SetRevertMessageToReceipt && receipt.Status == uint64(iotextypes.ReceiptStatus_ErrExecutionReverted) && retval != nil {
+		if featureCtx.FixCustomErrorRevertMessage {
+			reason, errUnpack := abi.UnpackRevert(retval)
+			//we need to catch custom errors
+			if errUnpack != nil {
+				reason = "execution reverted"
+			}
+			receipt.SetExecutionRevertMsg(reason)
```

Comment:
looks like similar code with `newRevertError`, why is `abi.UnpackRevert()` called twice

Reply:
here is only a check  whether it contains a custom error. Previously, an error was returned, but now it only catches the error without returning it. The specific error is converted into the  interface required by the JSON-RPC  according to the `newRevertError`